modules/uhttp/uhttpclient/client_middleware.go

```diff
@@ -85,9 +86,9 @@ func tracingOutbound(tracer opentracing.Tracer) OutboundMiddlewareFunc {
 
 // authenticationOutbound on client side calls authenticate, and gets a claim that client is who they say they are
 // We only authorize with the claim on server side
-func authenticationOutbound(info auth.CreateAuthInfo) OutboundMiddlewareFunc {
-	authClient := auth.Load(info)
-	serviceName := info.Config().Get(config.ServiceNameKey).AsString()
+func authenticationOutbound(cfg config.Provider, scope tally.Scope) OutboundMiddlewareFunc {
```

Comment:
Can we accept an auth.Client, instead of loading a new client for each middleware?